src/Avalonia.SceneGraph/Rendering/Renderer.cs

```diff
@@ -0,0 +1,67 @@
+﻿// Copyright (c) The Avalonia Project. All rights reserved.
+// Licensed under the MIT license. See licence.md file in the project root for full license information.
+
+using System;
+using Avalonia.Platform;
+using Avalonia.VisualTree;
+
+namespace Avalonia.Rendering
+{
+    public class Renderer : IDisposable, IRenderer
+    {
+        private IRenderLoop _renderLoop;
+        private IRenderRoot _root;
+        private IRenderTarget _renderTarget;
+        private bool _dirty;
+
+        public void Attach(IRenderRoot root, IRenderLoop renderLoop)
+        {
+            Contract.Requires<ArgumentNullException>(root != null);
```

Comment:
Nope, there is one renderer per top level control. I use `Attach` because it's created with the service locator and there's no way to pass constructor arguments using that.